resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -782,16 +782,16 @@ private[spark] class MesosClusterScheduler(
     retryState: Option[MesosClusterRetryState], status: TaskStatus): MesosClusterRetryState = {
 
     val (retries, waitTimeSec) = retryState
-      .map { rs => (rs.retries + 1, rs.waitTime) }
+      .map { rs => (rs.retries + 1, rs.waitTime * 2) }
       .getOrElse{ (1, 1) }
 
     // if a node is draining, the driver should be relaunched without backoff
     if (isNodeDraining(status)) {
-      new MesosClusterRetryState(status, retries, new Date(), waitTimeSec)
+      new MesosClusterRetryState(status, retries, new Date(),
+        if (waitTimeSec > 1) waitTimeSec / 2 else 1) // Keep waitTime the same
```

Comment:
Essentially, it looks like we roll back the change we applied on line 785 here so that we don't change the value on line 792. I can't see what are the benefits of this change? Also, reverting previously calculated exponential backoff doesn't seem to be a sound design decision.

Reply:
I Dropped the revert division part and rewrote it to be more clear.

The bug was that in case of an empty `retryState` (i.e. it's a first try), the initial waitTime should be kept as 1 second. An already existing test expected that and had a 1.5 sec delay. The node draining logic that was added, accidentally doubled the initial waitTime to 2 seconds causing that test to fail.

Reply:
got it, thanks for the explanation!